Running Jenkins with JDK 11 correctly displays the folder with `/` but running with JDK 8 crashes with error
```
javax.servlet.ServletException: org.apache.commons.jelly.JellyTagException: file:/Users/calvin/GitRepos/jenkins/core/src/main/resources/hudson/model/View/index.jelly:42:43: <st:include> org.apache.commons.jelly.JellyTagException: file:/Users/calvin/GitRepos/jenkins/core/src/main/resources/lib/hudson/projectView.jelly:84:48: <j:forEach> java.nio.CharBuffer.rewind()Ljava/nio/CharBuffer;
Caused by: org.apache.commons.jelly.JellyTagException: file:/Users/calvin/GitRepos/jenkins/core/src/main/resources/hudson/model/View/index.jelly:42:43: <st:include> org.apache.commons.jelly.JellyTagException: file:/Users/calvin/GitRepos/jenkins/core/src/main/resources/lib/hudson/projectView.jelly:84:48: <j:forEach> java.nio.CharBuffer.rewind()Ljava/nio/CharBuffer;
Caused by: java.lang.RuntimeException: org.apache.commons.jelly.JellyTagException: file:/Users/calvin/GitRepos/jenkins/core/src/main/resources/lib/hudson/projectView.jelly:84:48: <j:forEach> java.nio.CharBuffer.rewind()Ljava/nio/CharBuffer;
Caused by: org.apache.commons.jelly.JellyTagException: file:/Users/calvin/GitRepos/jenkins/core/src/main/resources/lib/hudson/projectView.jelly:84:48: <j:forEach> java.nio.CharBuffer.rewind()Ljava/nio/CharBuffer;
Caused by: java.lang.NoSuchMethodError: java.nio.CharBuffer.rewind()Ljava/nio/CharBuffer;
```

--- a/test/src/test/java/hudson/model/ItemGroupMixInTest.java
+++ b/test/src/test/java/hudson/model/ItemGroupMixInTest.java
@@ -216,12 +216,8 @@ public class ItemGroupMixInTest {
 
     Project goodProject = r.jenkins.createProject(FreeStyleProject.class, goodName);
 
-    try {
-      r.jenkins.copy(goodProject, badName);
-      fail("@ is an unsafe character therefore copying to a project named " + badName + " should have failed");
-    } catch (Failure x) {
-      assertEquals(x.getMessage(), Messages.Hudson_UnsafeChar("@"));
-    }
+    Failure exception = assertThrows(Failure.class, () -> { r.jenkins.copy(goodProject, badName); });
+    assertEquals(exception.getMessage(), Messages.Hudson_UnsafeChar("@"));
   }
 
   @Issue("JENKINS-61956")
@@ -229,12 +225,8 @@ public class ItemGroupMixInTest {
   public void createProject_checkGoodName() throws Failure, IOException {
     final String badName = "calvin@jenkins";
 
-    try {
-      r.jenkins.createProject(MockFolder.class, badName);
-      fail("@ is an unsafe character therefore creating a folder named " + badName + " should have failed");
-    } catch (Failure x) {
-      assertEquals(x.getMessage(), Messages.Hudson_UnsafeChar("@"));
-    }
+    Failure exception = assertThrows(Failure.class, () -> { r.jenkins.createProject(MockFolder.class, badName); });
+    assertEquals(exception.getMessage(), Messages.Hudson_UnsafeChar("@"));
   }
 
   @Issue("JENKINS-61956")
@@ -259,11 +251,9 @@ public class ItemGroupMixInTest {
             "  <buildWrappers/>\n" +
             "</project>";
 
-    try {
+    Failure exception = assertThrows(Failure.class, () -> {
       r.jenkins.createProjectFromXML(badName, new ByteArrayInputStream(xml.getBytes()));
-      fail("@ is an unsafe character therefore creating a project named " + badName + " should have failed");
-    } catch (Failure x) {
-      assertEquals(x.getMessage(), Messages.Hudson_UnsafeChar("@"));
-    }
+    });
+    assertEquals(exception.getMessage(), Messages.Hudson_UnsafeChar("@"));
   }
 }

--- a/test/src/test/java/hudson/model/ItemGroupMixInTest.java
+++ b/test/src/test/java/hudson/model/ItemGroupMixInTest.java
@@ -53,6 +53,7 @@ import java.util.Collection;
 import java.util.List;
 
 import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.assertEquals;
 
 public class ItemGroupMixInTest {
 
@@ -207,4 +208,62 @@ public class ItemGroupMixInTest {
         assertThat(Items.getConfigFile(foo).asString(), containsString("<description/>"));
     }
 
+  @Issue("JENKINS-61956")
+  @Test
+  public void copy_checkGoodName() throws Failure, IOException {
+    final String goodName = "calvin-jenkins";
+    final String badName = "calvin@jenkins";
+
+    Project goodProject = r.jenkins.createProject(FreeStyleProject.class, goodName);
+
+    try {
+      r.jenkins.copy(goodProject, badName);
+      fail("@ is an unsafe character therefore copying to a project named " + badName + " should have failed");
+    } catch (Failure x) {
+      assertEquals(x.getMessage(), Messages.Hudson_UnsafeChar("@"));
+    }
+  }
+
+  @Issue("JENKINS-61956")
+  @Test
+  public void createProject_checkGoodName() throws Failure, IOException {
+    final String badName = "calvin@jenkins";
+
+    try {
+      r.jenkins.createProject(MockFolder.class, badName);
+      fail("@ is an unsafe character therefore creating a folder named " + badName + " should have failed");
+    } catch (Failure x) {
+      assertEquals(x.getMessage(), Messages.Hudson_UnsafeChar("@"));
+    }
+  }
+
+  @Issue("JENKINS-61956")
+  @Test
+  public void createProjectFromXML_checkGoodName() throws Failure, IOException {
+    final String badName = "calvin@jenkins";
+
+    final String xml = "<?xml version='1.0' encoding='UTF-8'?>\n" +
+            "<!DOCTYPE project[\n" +
+            "  <!ENTITY foo SYSTEM \"file:///\">\n" +
+            "]>\n" +
+            "<project>\n" +
+            "  <actions/>\n" +
+            "  <description>&foo;</description>\n" +
+            "  <keepDependencies>false</keepDependencies>\n" +
+            "  <properties/>\n" +
+            "  <scm class=\"hudson.scm.NullSCM\"/>\n" +
+            "  <canRoam>true</canRoam>\n" +
+            "  <triggers/>\n" +
+            "  <builders/>\n" +
+            "  <publishers/>\n" +
+            "  <buildWrappers/>\n" +
+            "</project>";
+
+    try {
+      r.jenkins.createProjectFromXML(badName, new ByteArrayInputStream(xml.getBytes()));
+      fail("@ is an unsafe character therefore creating a project named " + badName + " should have failed");
+    } catch (Failure x) {
+      assertEquals(x.getMessage(), Messages.Hudson_UnsafeChar("@"));
+    }
+  }
 }

--- a/test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java
+++ b/test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java
@@ -97,7 +97,7 @@ public class ReverseBuildTriggerTest {
                 .grant(Computer.BUILD).everywhere().to("alice", "bob")
                 .grant(Jenkins.ADMINISTER).everywhere().to("admin");
         r.jenkins.setAuthorizationStrategy(auth);
-        String upstreamName = "upstr3@m"; // do not clash with English messages!
+        String upstreamName = "upstr3am"; // do not clash with English messages!
         final FreeStyleProject upstream = r.createFreeStyleProject(upstreamName);
         String downstreamName = "d0wnstr3am";
         FreeStyleProject downstream = r.createFreeStyleProject(downstreamName);

